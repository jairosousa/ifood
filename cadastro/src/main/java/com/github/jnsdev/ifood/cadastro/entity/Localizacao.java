package com.github.jnsdev.ifood.cadastro.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;

/**
 * @Autor Jairo Nascimento
 * @Created 24/11/2021 - 17:51
 */

@Entity
@Table(name = "localizacao")
public class Localizacao extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long id;

    public Double latitude;

    public Double longityde;

}
