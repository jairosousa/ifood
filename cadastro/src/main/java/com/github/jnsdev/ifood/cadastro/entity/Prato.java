package com.github.jnsdev.ifood.cadastro.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;
import java.math.BigDecimal;

/**
 * @Autor Jairo Nascimento
 * @Created 24/11/2021 - 17:54
 */
@Entity
@Table(name = "localizacao")
public class Prato extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long id;

    public String nome;

    public String descricao;

    @ManyToOne
    public Restaurante restaurante;

    public BigDecimal preco;
}
