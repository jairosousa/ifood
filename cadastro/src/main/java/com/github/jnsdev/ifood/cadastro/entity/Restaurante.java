package com.github.jnsdev.ifood.cadastro.entity;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

/**
 * @Autor Jairo Nascimento
 * @Created 24/11/2021 - 17:47
 */
@Entity
@Table(name = "restaurante")
public class Restaurante extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long id;

    public String proprietario;

    public String cnpj;

    public String nome;

    @ManyToOne
    public Localizacao localizacao;

    @CreationTimestamp
    public Date dataCriação;

    @UpdateTimestamp
    public Date dataAtualizacao;
}
