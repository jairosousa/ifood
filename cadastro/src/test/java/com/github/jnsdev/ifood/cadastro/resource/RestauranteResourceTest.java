package com.github.jnsdev.ifood.cadastro.resource;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.jnsdev.ifood.cadastro.CadastroTestLifecycleManager;
import com.github.jnsdev.ifood.cadastro.entity.Restaurante;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response;

/**
 * @Autor Jairo Nascimento
 * @Created 27/11/2021 - 20:45
 */
@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    void testBuscarRestaurantes() {
        String resultado = given()
                .when()
                .get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(resultado);
    }

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testAlterarRestaurante() {
        Restaurante dto = new Restaurante();
        dto.nome = "novoNome";
        Long parameterValue = 123L;

        given()
                .with().pathParam("id", parameterValue)
                .body(dto)
                .when()
                .put("/restaurantes/{id}")
                .then()
                .statusCode(Response.Status.NO_CONTENT.getStatusCode())
                .extract().asString();

        Restaurante byId = Restaurante.findById(parameterValue);
        Assert.assertEquals(dto.nome, byId.nome);
    }

    private RequestSpecification given() {
        return RestAssured.given().contentType(ContentType.JSON);
    }
}