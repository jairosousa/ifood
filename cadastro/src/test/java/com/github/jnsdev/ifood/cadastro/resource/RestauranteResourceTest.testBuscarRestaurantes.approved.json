[
  {
    "id": 123,
    "nome": "Novo Restaurante",
    "proprietario": "ID do keyclok"
  },
  {
    "id": 234,
    "nome": "Novo Restaurante 2",
    "proprietario": "ID do keyclok 2"
  }
]